streams/src/test/java/org/apache/kafka/test/GenericInMemoryKeyValueStore.java

```diff
@@ -16,24 +16,28 @@
  */
 package org.apache.kafka.test;
 
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.util.Map.Entry;
-import java.util.NavigableMap;
-import java.util.TreeMap;
 import org.apache.kafka.streams.KeyValue;
 import org.apache.kafka.streams.processor.ProcessorContext;
 import org.apache.kafka.streams.processor.StateStore;
 import org.apache.kafka.streams.state.KeyValueIterator;
 import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.internals.CacheFlushListener;
+import org.apache.kafka.streams.state.internals.CachedStateStore;
 import org.apache.kafka.streams.state.internals.DelegatingPeekingKeyValueIterator;
 
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+
 /**
  * This class is a generic version of the in-memory key-value store that is useful for testing when you
  *  need a basic KeyValueStore for arbitrary types and don't have/want to write a serde
  */
-public class GenericInMemoryKeyValueStore<K extends Comparable, V> implements KeyValueStore<K, V> {
+public class GenericInMemoryKeyValueStore<K extends Comparable, V>
+    implements KeyValueStore<K, V>, CachedStateStore<K, V> {
```

Comment:
Because we assume that all "metered stores" implement `CachedStateStore` we need to add it here, too.

Reply:
Sorry, I couldn't follow... How is this a "metered store"?

Was it necessary to add the interface to get some tests to work?

Reply:
It's not a metered store, but we assume a certain wrapping hierarchy and `GenericInMemoryKeyValueStore` is in the same hierarchy level as "metered stores" -- for this test case, there is no metered stores, but `GenericInMemoryKeyValueStore` takes the metered stores place in the hierarchy. So, yes, it required to get the test work.